ambry-api/src/main/java/com/github/ambry/compression/Compression.java

```diff
@@ -92,21 +102,23 @@ public interface Compression {
    * The compressed buffer may be shared with other buffers, meaning only portion of this buffer was used by
    * compression.  That is why it accepts the offset and size parameters.
    * @param compressedBuffer The compressed buffer.
-   * @param offset The offset in the compressed buffer where compression data is stored.
-   * @param size The size in compressedBuffer to read.
+   * @param compressedBufferOffset The offset in the compressed buffer where compression data is stored.
+   * @param compressedDataSize The size in compressedBuffer to read, may not be the size of compressedBuffer.
    * @return The size of original data.
    */
-  int getDecompressBufferSize(byte[] compressedBuffer, int offset, int size);
+  int getDecompressBufferSize(byte[] compressedBuffer, int compressedBufferOffset, int compressedDataSize)
+      throws CompressionException;
 
   /**
    * Get the original data size stored in compressed buffer.  Compressed buffer is not shared and contains
    * compressed data only.  The compressed buffer is the output of compress() that contains original/source data size.
-   * @param compressedBuffer The compressed buffer.
+   * @param compressedBuffer The compressed buffer.  It cannot be null or empty.
    * @return Size of the original data.
+   * @throws CompressionException Throws this exception if compression has internal failure.
    */
-  default int getDecompressBufferSize(byte[] compressedBuffer) {
-    if (compressedBuffer == null) {
-      throw new IllegalArgumentException("compressedBuffer cannot be null.");
+  default int getDecompressBufferSize(byte[] compressedBuffer) throws CompressionException {
+    if (compressedBuffer == null || compressedBuffer.length == 0) {
```

Comment:
condense into a helper like isCompressedBufferEmpty()

Reply:
Even better, I'd add a method to Utils just to verify empty byte array.  Will appear in next update.